python/test/test_polars.py

```diff
@@ -692,4 +692,52 @@ def test_csv_bad_encoding_loading():
             {"name": [name]},
             schema={"name": pl.String},
         )
-        pl_testing.assert_frame_equal(df, expected)
\ No newline at end of file
+        pl_testing.assert_frame_equal(df, expected)
+
+
+def test_categorical_domain():
+    pl = pytest.importorskip("polars")
+
+    lf_domain = dp.lazyframe_domain([dp.series_domain("A", dp.categorical_domain())])
+    assert str(lf_domain) == "FrameDomain(A: cat; margins=[])"
+
+    lf_domain = dp.lazyframe_domain([dp.series_domain("A", dp.option_domain(dp.categorical_domain()))])
+    assert str(lf_domain) == "FrameDomain(A: cat; margins=[])"
+
+    lf = pl.LazyFrame([pl.Series("A", ["Texas", "New York", None], dtype=pl.Categorical)])
+
+    with pytest.raises(dp.OpenDPException):
```

Comment:
Also check the error message? Want to be sure that it makes sense to the user.

(I also see elsewhere in this PR that the categorical is wrapped by optional... but I guess that doesn't mean we can fill it with nulls? -- If that does enable anything, special that too.)

Reply:
I explicitly check both now.